fiftyone/brain/internal/core/pinecone.py

```diff
@@ -209,30 +229,76 @@ def _initialize(self):
             self.save_config()
 
         if self.config.index_name in index_names:
-            index = pinecone.Index(self.config.index_name)
+            index = (
+                self._pinecone.Index(self.config.index_name)
+                if self._pinecone
+                else pinecone.Index(self.config.index_name)
+            )
         else:
             index = None
 
         self._index = index
 
     def _create_index(self, dimension):
-        kwargs = dict(
-            index_type=self.config.index_type,
-            metric=self.config.metric,
-            replicas=self.config.replicas,
-            shards=self.config.shards,
-            pods=self.config.pods,
-            pod_type=self.config.pod_type,
-        )
-        kwargs = {k: v for k, v in kwargs.items() if v is not None}
-
-        pinecone.create_index(
-            self.config.index_name,
-            dimension=dimension,
-            **kwargs,
-        )
+        if self._pinecone:
+            if self.config.index_type in [None, "serverless"]:
+                from pinecone import ServerlessSpec
+
+                self._pinecone.create_index(
+                    name=self.config.index_name,
+                    dimension=dimension,
+                    metric=self.config.metric
+                    if self.config.metric
+                    else "cosine",
+                    spec=ServerlessSpec(
+                        cloud=self.config.cloud,
+                        region=self.config.environment,
+                    ),
+                )
+            elif self.config.index_type == "pod":
+                from pinecone import PodSpec
 
-        self._index = pinecone.Index(self.config.index_name)
+                pod_type = (
+                    self.config.pod_type if self.config.pod_type else "p1.x1"
+                )
+                pods = self.config.pods if self.config.pods else 1
+                replicas = self.config.replicas if self.config.replicas else 1
+                shards = self.config.shards if self.config.shards else 1
+                self._pinecone.create_index(
+                    name=self.config.index_name,
+                    dimension=dimension,
+                    metric=self.config.metric
+                    if self.config.metric
+                    else "cosine",
+                    spec=PodSpec(
+                        environment=self.config.environment,
+                        pod_type=pod_type,
+                        pods=pods,
+                        replicas=replicas,
+                        shards=shards,
+                    ),
+                )
+            while not self._pinecone.describe_index(
+                self.config.index_name
+            ).status["ready"]:
+                time.sleep(1)
```

Comment:
Yes, `pinecone` takes times to initialize and I was running into issues because of that. Agree with the no timeout in core but I didn't find a method in pinecone SDK for this. I will check again to see if there's an alternative

Reply:
The MongoDB backend has a similar behavior so we added a `ready` property:
https://github.com/voxel51/fiftyone-brain/blob/2374ef5b20985ad65b38665d55effd1cce725a7a/fiftyone/brain/internal/core/mongodb.py#L282-L301

That can be used as documented here:
https://docs.voxel51.com/integrations/mongodb.html#checking-if-an-index-is-ready

Can you adopt that pattern here for consistency?